### PR TITLE
test: Dump coredumps from journal in the integration test wrapper

### PR DESCRIPTION
**Pull request by daandemeyer**
Fixes https://github.com/systemd/systemd/issues/35277

### CODE REVIEW

File: test/integration-test-wrapper.py [hunk 1]
```diff
@@ -34,6 +35,55 @@
...
 
 
+def dump_coredumps(args: argparse.Namespace, journal_file: Path) -> bool:
+    # Collect executable paths of all coredumps and filter out the expected ones.
+    # The following are excluded:
+    #   sleep/bash - intentional SIGABRT caused by TEST-57
+    #   systemd-notify - intermittent (and intentional) SIGABRT caused by TEST-59
+    #   test-execute - intentional coredump in TEST-02
+    #   test(-usr)?-dump - intentional coredumps from systemd-coredump tests in TEST-74
+    exclude_regex = re.compile("/(bash|sleep|systemd-notify|test-execute|test(-usr)?-dump)")
```
Comment (yuwata) [inline test/integration-test-wrapper.py]: Maybe, it is better to allow to append or override this by each test.
See https://github.com/systemd/systemd-centos-ci/blob/86adb3039c359214eccafa933e71ca001723143f/common/utils.sh#L10-L21 .

**Review by yuwata — COMMENTED** [review]
Thank you!

File: test/integration-test-wrapper.py (hunk 1 above)
Comment (DaanDeMeyer) [reply test/integration-test-wrapper.py:45]: @yuwata I would see if there is a need first, we'll find out as soon as the integration tests finish

**Comment by DaanDeMeyer** [comment]
This now has a bunch of assorted fixes as well, @yuwata PTAL!

**Comment by DaanDeMeyer** [comment]
The only remaining failure is

```
            PID: 12073 (systemd-localed)
           UID: 0 (root)
           GID: 0 (root)
        Signal: 6 (ABRT)
     Timestamp: Sat 2024-11-23 23:57:00 UTC (3min 7s ago)
  Command Line: /usr/lib/systemd/systemd-localed
    Executable: /usr/lib/systemd/systemd-localed
 Control Group: /system.slice/systemd-localed.service
          Unit: systemd-localed.service
         Slice: system.slice
       Boot ID: 04a2028b4ee2490c84492e85f6f21481
    Machine ID: 7b37e13a88a74d0998e4ce5c8e1ac8bc
      Hostname: H
       Storage: journal
       Message: Process 12073 (systemd-localed) of user 0 dumped core.
                
                Module libxkbcommon.so.0 from rpm libxkbcommon-1.7.0-4.fc41.x86_64
                Module libcap-ng.so.0 from rpm libcap-ng-0.8.5-3.fc41.x86_64
                Module libpcre2-8.so.0 from rpm pcre2-10.44-1.fc41.1.x86_64
                Module libeconf.so.0 from rpm libeconf-0.7.4-3.fc42.x86_64
                Module libaudit.so.1 from rpm audit-4.0.2-1.fc41.x86_64
                Module libz.so.1 from rpm zlib-ng-2.2.2-1.fc42.x86_64
                Module libattr.so.1 from rpm attr-2.5.2-4.fc41.x86_64
                Module libselinux.so.1 from rpm libselinux-3.7-6.fc42.x86_64
                Module libseccomp.so.2 from rpm libseccomp-2.5.5-2.fc41.x86_64
                Module libpam.so.0 from rpm pam-1.7.0-2.fc42.x86_64
                Module libcrypto.so.3 from rpm openssl-3.2.2-8.fc42.x86_64
                Module libmount.so.1 from rpm util-linux-2.40.2-8.fc42.x86_64
                Module libcrypt.so.2 from rpm libxcrypt-4.4.36-10.fc42.x86_64
                Module libcap.so.2 from rpm libcap-2.71-1.fc42.x86_64
                Module libblkid.so.1 from rpm util-linux-2.40.2-8.fc42.x86_64
                Module libacl.so.1 from rpm acl-2.3.2-2.fc41.x86_64
                Stack trace of thread 12073:
                #0  0x00007f47806e768b __sched_yield (libc.so.6 + 0xdb68b)
                #1  0x00007f478181f6d5 _ZN11__sanitizer12StopTheWorldEPFvRKNS_20SuspendedThreadsListEPvES3_ (libclang_rt.asan.so + 0x1f6d5)
                #2  0x00007f478183106e _ZN6__lsanL32LockStuffAndStopTheWorldCallbackEP12dl_phdr_infomPv (libclang_rt.asan.so + 0x3106e)
                #3  0x00007f478074e3d9 dl_iterate_phdr (libc.so.6 + 0x1423d9)
                #4  0x00007f4781831043 _ZN6__lsan24LockStuffAndStopTheWorldEPFvRKN11__sanitizer20SuspendedThreadsListEPvEPNS_18CheckForLeaksParamE (libclang_rt.asan.so + 0x31043)
                #5  0x00007f478182d121 _ZN6__lsanL13CheckForLeaksEv (libclang_rt.asan.so + 0x2d121)
                #6  0x00007f478182cecb _ZN6__lsan11DoLeakCheckEv (libclang_rt.asan.so + 0x2cecb)
                #7  0x00007f4780628540 __cxa_finalize (libc.so.6 + 0x1c540)
                #8  0x00007f4781805ed7 __do_global_dtors_aux (libclang_rt.asan.so + 0x5ed7)
                #9  0x00007f4782364092 _dl_call_fini (ld-linux-x86-64.so.2 + 0x2092)
                #10 0x00007f478236722e _dl_fini (ld-linux-x86-64.so.2 + 0x522e)
                #11 0x00007f4780628ac1 __run_exit_handlers (libc.so.6 + 0x1cac1)
                #12 0x00007f4780628b9e exit (libc.so.6 + 0x1cb9e)
                #13 0x00007f47806100cf __libc_start_call_main (libc.so.6 + 0x40cf)
                #14 0x00007f478061018b __libc_start_main@@GLIBC_2.34 (libc.so.6 + 0x418b)
                #15 0x0000561991d944d5 _start (/usr/lib/systemd/systemd-localed + 0xa4d5)
                ELF object binary architecture: AMD x86-64
```

Which happens because stopping localed with `SIGTERM` times out and pid 1 then sends `SIGABRT`. @mrc0mmand did you ever had to deal with services being too slow to stop because of leak sanitizer?

**Comment by DaanDeMeyer** [comment]
And opened a tracking issue: https://github.com/systemd/systemd/issues/35335

File: src/coredump/coredumpctl.c [hunk 2]
```diff
@@ -965,7 +965,9 @@ static int dump_list(int argc, char **argv, void *userdata) {
                 if (!arg_field && n_found <= 0) {
                         if (!arg_quiet)
                                 log_notice("No coredumps found.");
-                        return -ESRCH;
```
Comment (bluca) [inline src/coredump/coredumpctl.c]: Can we have a workaround in the script instead, at least for 257? That way we can already merge this, and change this return value in 258 and remove the workaround

**Review by bluca — APPROVED** [review]

**Comment by yuwata** [comment]
Actually it seems work: https://github.com/systemd/systemd/actions/runs/12004745750/job/33460209340?pr=35342
Thank you!

File: test/integration-test-wrapper.py [hunk 3]
```diff
@@ -32,6 +33,59 @@
...
+            '--directory', os.fspath(args.meson_source_dir),
+            '--extra-search-path', os.fspath(args.meson_build_dir),
+            'sandbox',
+            'coredumpctl',
+            '--file', journal_file,
+            '--json=short',
+        ],
+        stdout=subprocess.PIPE,
+        text=True,
+    )
```
Comment (yuwata) [inline test/integration-test-wrapper.py]:
```diff
diff --git a/test/integration-test-wrapper.py b/test/integration-test-wrapper.py
index 775ba3d39d..a3f90dc9fa 100755
--- a/test/integration-test-wrapper.py
+++ b/test/integration-test-wrapper.py
@@ -53,7 +53,7 @@ def process_coredumps(args: argparse.Namespace, journal_file: Path) -> bool:
         ],
         stdout=subprocess.PIPE,
         text=True,
-    )
+    )  # fmt: skip
 
     # coredumpctl returns a non-zero exit status if there are no coredumps.
     if result.returncode != 0:
```

**Review by yuwata — APPROVED** [review]

**Comment by mrc0mmand** [comment]
> in TEST-73-LOCALE which happens because stopping localed with `SIGTERM` times out and pid 1 then sends `SIGABRT`. @mrc0mmand did you ever had to deal with services being too slow to stop because of leak sanitizer?

I've never seen this one myself before.

**Comment by evverx** [comment]
I'm not sure it works when `journald` itself crashes. Its backtraces can end up in the journal but those messages don't have the message ids `coredumpctl` looks for. I've just triggered https://github.com/evverx/systemd/actions/runs/12063527181/job/33638872309?pr=114 (where `journald` is crashed in TEST-04-JOURNAL) to see what happens.

(On a somewhat related note inspired by that BOOTP issue where `mkosi` is used to test the BOOTP client I think it would be cool if coredumps were reported when, say, containers run with `mkosi boot` are stopped. It should make it harder to miss crashes with all those autorestarts. The bash framework can't do that (but it can show ASan backtraces when `make run` stops in its "manual" mode))

**Pull request merged**